modules/wallet/defrag.go

```diff
@@ -130,9 +129,17 @@ func (w *Wallet) threadedDefragWallet() {
 		w.mu.Unlock()
 		return
 	}
+	w.mu.Unlock()
+
+	// get the required fee for the defrag transaction.  35 outputs at an
+	// estimated 250 bytes needed per output means about a 10kb total
+	// transaction.
+	sizeAvgOutput := uint64(250) // bytes
+	fee := w.defragFee(sizeAvgOutput * defragBatchSize)
 
 	// Create the defrag transaction.
-	txnSet, err := w.createDefragTransaction()
+	w.mu.Lock()
+	txnSet, err := w.createDefragTransaction(fee)
```

Comment:
why is the fee calculated outside `createDefragTransaction`? Wouldn't the function be in a better position to calculate the fee, since it knows how many outputs it's defragging?

Reply:
the fee is calculated outside of `createDefragTransaction` because fee calculation requires calling an exported method (`tpool.FeeEstimation()`), and `createDefragTransaction` is called under the wallet's lock.

Reply:
agreed, `managedCreateDefragTransaction` seems fine to me